shared/multisig.py

```diff
@@ -889,10 +889,94 @@ def update_contents(self):
         MenuItem('Delete', f=ms_wallet_delete, arg=ms),
         MenuItem('Coldcard Export', f=ms_wallet_ckcc_export, arg=ms),
         MenuItem('Electrum Wallet', f=ms_wallet_electrum_export, arg=ms),
+        MenuItem('Address Explorer', f=multisig_address_explorer, arg=ms)
     ]
 
     return rv
 
+def make_redeem(M, keys, path):
+    # Construct a redeem script, and ordered list of pubkeys
+    N = len(keys)
+
+    # derive pubkeys (see BIP 67: <https://github.com/bitcoin/bips/blob/master/bip-0067.mediawiki>)
+    pubkeys = []
+    for xfp, node in keys:
+        depth = node.depth()
+        for p in path[depth:]: # derive entire path (in-place)
+            node.derive(p)
+        pubkeys.append(node.public_key())
+    pubkeys.sort()
+
+    # construct redeem script
+    mm = [80 + M] if M <= 16 else [1, M]
+    nn = [80 + N] if N <= 16 else [1, N]
+    rv = bytes(mm)
+    for pk in pubkeys:
+        rv += bytes([len(pk)]) + pk
+    rv += bytes(nn + [0xAE])
+
+    return rv, pubkeys
+
+def ms_path_mapper(idx, is_change, coin, addr_fmt):
```

Comment:
Please don't pass around 'coin' ... instead, just lookup the current chain when needed. Affects this function and a few others nearby.

Reply:
Noted